### DISCUSSION TITLE
Infinity with Hasura GraphQL

### DISCUSSION THREAD

**Discussion by PhilippSchultze** [grafana/grafana-infinity-datasource#291] [Q&A]
Hello,
do you have any experiences using Hasura GraphQL? I am not able to run any successful query.
I always get response code 500 from the server. This seems like a communication problem between the data plugin and the Hasura GraphQL plugin.

Best regards.

**Comment by elasticdotventures** [comment]
@PhilippSchultze did you make any progress on this?

**Reply by PhilippSchultze** [reply]
I did not pursue the topic further. I wanted to try out your plugin as an option to bypass some existing bugs in the Grafana GraphQL Data Source plugin. Well, getting no data at all was even worse than having a few bugs with certain queries. :-)

**Reply by elasticdotventures** [reply]
Thanks @PhilippSchultze, I'm happy to do a write up and post examples.  I need to get Hasura + infinity working, either that or ditch the whole grafana approach in favor of something like apache superset. 

@yesoreyeram 🙏 any advice / tips?

Background:
Hasura is a popular middleware, which frequently provides a websocket facing graphql subscription & access control layer to databases like postgres.  

Where I am:
I've successfully gotten the JWT mechanism with Hasura to work, it's quite a few steps.  I have successfully made GraphQL queries, I'm presently using UQL to parse them, which is a bit laborious.   Presently - this is "blocked" and not functional beyond the MVP I just mentioned because the JWT must be extracted from the browser of another application and then paste into the Grafana infinity data source, saved, & it is shared/accessible to all users, and it expires after a while. 

What I'm trying to do:
My requirements need individual-user issued credentials to pass through Grafana (via infinity) to Hasura. 


What I think the problem is:
Hasura is a bit confusing, because it talks about OAuth blabla, but it uses Auth0 to issue a JWT. Hasura only seems to use the JWT. 

We use the same Auth0 SSO mechanism to let users login to Grafana and "silly me" thought the OAuth passThrough mode would take a JWT issued to Grafana and pass it through to Hasura (and this would be easy, peasy lemon squeezy)!  nope. that is not how it works.😿

any help or suggestions is appreciated.

**Reply by elasticdotventures** [reply]
As an update - upon further investigation Grafana does have a JWT Authentication type
https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/jwt/
